From 0e91e3cca268d6d7e640455eba30003297f2747a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Yilli=C3=A9?= <yilliee@protonmail.com>
Date: Sun, 3 Oct 2021 18:39:40 +0500
Subject: [PATCH 2/3] Add an option to specify a custom x-axis value for status
 bar items
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

 - The var TW_CUSTOM_CLOCK_POS, TW_CUSTOM_CPU_POS and
   TW_CUSTOM_BATTERY_POS can be used to specify a custom position
   for the clock, cpu temperature and battery percentage on the x-axis.
 - One of the use-cases of this var is when the status bar items overlap
   with the camera cutout. In that case, instead of specifiying an offset
   for the whole gui on y-axis, the status bar items can be offset along
   the x-axis only.

Signed-off-by: Yillié <yilliee@protonmail.com>
Change-Id: Ic01d301731822a598b0ca8adb2280c4d01a3622f
Signed-off-by: Mohd Faraz <androiabledroid@gmail.com>
---
 gui/libguitwrp_defaults.go      | 14 ++++++++++++++
 gui/theme/landscape_hdpi/ui.xml |  9 ++++++---
 gui/theme/landscape_mdpi/ui.xml |  9 ++++++---
 gui/theme/portrait_hdpi/ui.xml  |  9 ++++++---
 gui/theme/portrait_mdpi/ui.xml  |  9 ++++++---
 gui/theme/watch_mdpi/ui.xml     |  9 ++++++---
 6 files changed, 44 insertions(+), 15 deletions(-)

diff --git a/gui/libguitwrp_defaults.go b/gui/libguitwrp_defaults.go
index 222564c7f..258ac8daf 100644
--- a/gui/libguitwrp_defaults.go
+++ b/gui/libguitwrp_defaults.go
@@ -76,6 +76,15 @@ func copyThemeResources(ctx android.BaseContext, dirs []string, files []string)
 			version = strings.Split(line, " ")[2]
 		}
 	}
+
+	_props := [3]string{"TW_CUSTOM_BATTERY_POS", "TW_CUSTOM_CLOCK_POS", "TW_CUSTOM_CPU_POS"}
+	props := [3]string{"%indent_right%", "%center_x%", "%indent%"}
+	for i, item := range _props {
+		if getMakeVars(ctx, item) != "" {
+			props[i] = getMakeVars(ctx, item)
+		}
+	}
+
 	_files := [2]string{"splash.xml", "ui.xml"}
 	for _, i := range _files {
 		data, err = ioutil.ReadFile(twRes + i)
@@ -84,6 +93,11 @@ func copyThemeResources(ctx android.BaseContext, dirs []string, files []string)
 			return
 		}
 		newFile := strings.Replace(string(data), "{themeversion}", version, -1)
+		if i == "ui.xml" {
+			newFile = strings.Replace(newFile, "{battery_pos}", props[0], -1)
+			newFile = strings.Replace(newFile, "{clock_pos}", props[1], -1)
+			newFile = strings.Replace(newFile, "{cpu_pos}", props[2], -1)
+		}
 		err = ioutil.WriteFile(twRes + i, []byte(newFile), 0)
 		if err != nil {
 			fmt.Println(err)
diff --git a/gui/theme/landscape_hdpi/ui.xml b/gui/theme/landscape_hdpi/ui.xml
index bd055bd51..2f210f3e8 100755
--- a/gui/theme/landscape_hdpi/ui.xml
+++ b/gui/theme/landscape_hdpi/ui.xml
@@ -226,6 +226,9 @@
 		<variable name="pattern_dot_dia" value="32"/>
 		<variable name="pattern_line_w" value="12"/>
 		<variable name="pattern_size" value="648"/>
+		<variable name="tw_clock_pos_x" value="{clock_pos}"/>
+		<variable name="tw_cpu_pos_x" value="{cpu_pos}"/>
+		<variable name="tw_battery_pos_x" value="{battery_pos}"/>
 	</variables>
 
 	<mousecursor>
@@ -275,13 +278,13 @@
 			<text color="%text_color%">
 				<condition var1="tw_no_cpu_temp" var2="0"/>
 				<font resource="font_m"/>
-				<placement x="%indent%" y="%row1_header_y%"/>
+				<placement x="%tw_cpu_pos_x%" y="%row1_header_y%"/>
 				<text>{@cpu_temp=CPU: %tw_cpu_temp% °C}</text>
 			</text>
 
 			<text color="%text_color%">
 				<font resource="font_m"/>
-				<placement x="%center_x%" y="%row1_header_y%" placement="5"/>
+				<placement x="%tw_clock_pos_x%" y="%row1_header_y%" placement="5"/>
 				<text>%tw_time%</text>
 			</text>
 
@@ -292,7 +295,7 @@
 					<condition var1="tw_battery" op="&lt;" var2="101"/>
 				</conditions>
 				<font resource="font_m"/>
-				<placement x="%indent_right%" y="%row1_header_y%" placement="1"/>
+				<placement x="%tw_battery_pos_x%" y="%row1_header_y%" placement="1"/>
 				<text>{@battery_pct=Battery: %tw_battery%}</text>
 			</text>
 
diff --git a/gui/theme/landscape_mdpi/ui.xml b/gui/theme/landscape_mdpi/ui.xml
index 9ab8863d0..58a259b2e 100755
--- a/gui/theme/landscape_mdpi/ui.xml
+++ b/gui/theme/landscape_mdpi/ui.xml
@@ -226,6 +226,9 @@
 		<variable name="pattern_dot_dia" value="12"/>
 		<variable name="pattern_line_w" value="4"/>
 		<variable name="pattern_size" value="252"/>
+		<variable name="tw_clock_pos_x" value="{clock_pos}"/>
+		<variable name="tw_cpu_pos_x" value="{cpu_pos}"/>
+		<variable name="tw_battery_pos_x" value="{battery_pos}"/>
 	</variables>
 
 	<mousecursor>
@@ -275,13 +278,13 @@
 			<text color="%text_color%">
 				<condition var1="tw_no_cpu_temp" var2="0"/>
 				<font resource="font_m"/>
-				<placement x="%indent%" y="%row1_header_y%"/>
+				<placement x="%tw_cpu_pos_x%" y="%row1_header_y%"/>
 				<text>{@cpu_temp=CPU: %tw_cpu_temp% °C}</text>
 			</text>
 
 			<text color="%text_color%">
 				<font resource="font_m"/>
-				<placement x="%center_x%" y="%row1_header_y%" placement="5"/>
+				<placement x="%tw_clock_pos_x%" y="%row1_header_y%" placement="5"/>
 				<text>%tw_time%</text>
 			</text>
 
@@ -292,7 +295,7 @@
 					<condition var1="tw_battery" op="&lt;" var2="101"/>
 				</conditions>
 				<font resource="font_m"/>
-				<placement x="%indent_right%" y="%row1_header_y%" placement="1"/>
+				<placement x="%tw_battery_pos_x%" y="%row1_header_y%" placement="1"/>
 				<text>{@battery_pct=Battery: %tw_battery%}</text>
 			</text>
 
diff --git a/gui/theme/portrait_hdpi/ui.xml b/gui/theme/portrait_hdpi/ui.xml
index f7d21da3f..4eccbadbb 100755
--- a/gui/theme/portrait_hdpi/ui.xml
+++ b/gui/theme/portrait_hdpi/ui.xml
@@ -223,6 +223,9 @@
 		<variable name="pattern_dot_dia" value="32"/>
 		<variable name="pattern_line_w" value="12"/>
 		<variable name="pattern_size" value="828"/>
+		<variable name="tw_clock_pos_x" value="{clock_pos}"/>
+		<variable name="tw_cpu_pos_x" value="{cpu_pos}"/>
+		<variable name="tw_battery_pos_x" value="{battery_pos}"/>
 	</variables>
 
 	<mousecursor>
@@ -272,13 +275,13 @@
 			<text color="%text_color%">
 				<condition var1="tw_no_cpu_temp" var2="0"/>
 				<font resource="font_m"/>
-				<placement x="%indent%" y="%row1_header_y%"/>
+				<placement x="%tw_cpu_pos_x%" y="%row1_header_y%"/>
 				<text>{@cpu_temp=CPU: %tw_cpu_temp% °C}</text>
 			</text>
 
 			<text color="%text_color%">
 				<font resource="font_m"/>
-				<placement x="%center_x%" y="%row1_header_y%" placement="5"/>
+				<placement x="%tw_clock_pos_x%" y="%row1_header_y%" placement="5"/>
 				<text>%tw_time%</text>
 			</text>
 
@@ -289,7 +292,7 @@
 					<condition var1="tw_battery" op="&lt;" var2="101"/>
 				</conditions>
 				<font resource="font_m"/>
-				<placement x="%indent_right%" y="%row1_header_y%" placement="1"/>
+				<placement x="%tw_battery_pos_x%" y="%row1_header_y%" placement="1"/>
 				<text>{@battery_pct=Battery: %tw_battery%}</text>
 			</text>
 
diff --git a/gui/theme/portrait_mdpi/ui.xml b/gui/theme/portrait_mdpi/ui.xml
index 9189df0c7..42ead0561 100755
--- a/gui/theme/portrait_mdpi/ui.xml
+++ b/gui/theme/portrait_mdpi/ui.xml
@@ -223,6 +223,9 @@
 		<variable name="pattern_dot_dia" value="12"/>
 		<variable name="pattern_line_w" value="4"/>
 		<variable name="pattern_size" value="360"/>
+		<variable name="tw_clock_pos_x" value="{clock_pos}"/>
+		<variable name="tw_cpu_pos_x" value="{cpu_pos}"/>
+		<variable name="tw_battery_pos_x" value="{battery_pos}"/>
 	</variables>
 
 	<mousecursor>
@@ -272,13 +275,13 @@
 			<text color="%text_color%">
 				<condition var1="tw_no_cpu_temp" var2="0"/>
 				<font resource="font_m"/>
-				<placement x="%indent%" y="%row1_header_y%"/>
+				<placement x="%tw_cpu_pos_x%" y="%row1_header_y%"/>
 				<text>{@cpu_temp=CPU: %tw_cpu_temp% °C}</text>
 			</text>
 
 			<text color="%text_color%">
 				<font resource="font_m"/>
-				<placement x="%center_x%" y="%row1_header_y%" placement="5"/>
+				<placement x="%tw_clock_pos_x%" y="%row1_header_y%" placement="5"/>
 				<text>%tw_time%</text>
 			</text>
 
@@ -289,7 +292,7 @@
 					<condition var1="tw_battery" op="&lt;" var2="101"/>
 				</conditions>
 				<font resource="font_m"/>
-				<placement x="%indent_right%" y="%row1_header_y%" placement="1"/>
+				<placement x="%tw_battery_pos_x%" y="%row1_header_y%" placement="1"/>
 				<text>{@battery_pct=Battery: %tw_battery%}</text>
 			</text>
 
diff --git a/gui/theme/watch_mdpi/ui.xml b/gui/theme/watch_mdpi/ui.xml
index ed7ac825c..478509409 100644
--- a/gui/theme/watch_mdpi/ui.xml
+++ b/gui/theme/watch_mdpi/ui.xml
@@ -182,6 +182,9 @@
 		<variable name="pattern_dot_dia" value="12"/>
 		<variable name="pattern_line_w" value="4"/>
 		<variable name="pattern_size" value="200"/>
+		<variable name="tw_clock_pos_x" value="{clock_pos}"/>
+		<variable name="tw_cpu_pos_x" value="{cpu_pos}"/>
+		<variable name="tw_battery_pos_x" value="{battery_pos}"/>
 	</variables>
 
 	<templates>
@@ -211,13 +214,13 @@
 			<text color="%text_color%">
 				<condition var1="tw_no_cpu_temp" var2="0"/>
 				<font resource="font_m"/>
-				<placement x="%indent%" y="%row1_header_y%"/>
+				<placement x="%tw_battery_pos_x%" y="%row1_header_y%"/>
 				<text>{@cpu_temp=CPU: %tw_cpu_temp% °C}</text>
 			</text>
 
 			<text color="%text_color%">
 				<font resource="font_m"/>
-				<placement x="%center_x%" y="%row1_header_y%" placement="5"/>
+				<placement x="%tw_clock_pos_x%" y="%row1_header_y%" placement="5"/>
 				<text>%tw_time%</text>
 			</text>
 
@@ -228,7 +231,7 @@
 					<condition var1="tw_battery" op="&lt;" var2="101"/>
 				</conditions>
 				<font resource="font_m"/>
-				<placement x="%indent_right%" y="%row1_header_y%" placement="1"/>
+				<placement x="%tw_battery_pos_x%" y="%row1_header_y%" placement="1"/>
 				<text>{@battery_pct=Battery: %tw_battery%}</text>
 			</text>
 
-- 
2.36.1

